File: countingsort.py
arr = [1,4,1,2,7,5,2]
tamano_arreglo = len(arr)
arr_salida = []
cumulo = []

maximo_elemento_arr = 0

for i in arr:
  if (maximo_elemento_arr < i):
    maximo_elemento_arr = i

for i in range(0, maximo_elemento_arr + 1):
  cumulo.append(0)

for i in arr:
  cumulo[i] += 1

for i in range(1, maximo_elemento_arr + 1):
  cumulo[i] += cumulo[i - 1]

for i in range(0, tamano_arreglo):
  arr_salida.append(0)

for i in arr:
  indice_salida = cumulo[i] - 1
  arr_salida[indice_salida] = i
  cumulo[i] -= 1

for i in arr_salida:
  print(i, end=" ")

File: countingsort.cpp
#include <iostream>

int main() {
  
  int arr[] = {1,4,1,2,7,5,2};
  int tamano_arr = sizeof(arr) / sizeof(arr[0]);
  int i, max_value = 0;
  int arr_salida[tamano_arr];
  
  for (i = 0; i < tamano_arr; i++)
    if (max_value < arr[i]) max_value = arr[i];

  int cumulo[max_value + 1];
  
  for (i = 0; i < max_value + 1; i++) cumulo[i] = 0;

  for (i = 0; i < tamano_arr; i++)
    cumulo[arr[i]]++;

  for (i = 1; i < max_value + 1; i++) {
    cumulo[i] += cumulo[i - 1];
  }
  
  int indice_salida = 0;
  for (i = 0; i < tamano_arr; i++) {
    indice_salida = cumulo[arr[i]] - 1;
    arr_salida[indice_salida] = arr[i];
    cumulo[arr[i]] -= 1;
  }

  for (i = 0; i < tamano_arr; i++) std::cout << arr_salida[i] << " ";
  std::cout << std::endl;
}